### PR TITLE
fix(Tabs): refactoring tabs selection

### LINKED ISSUE #5471: Incorrect selection indicator after page routing
## Description  
Incorrect selection indicator is left after navigating to the root URL of a page.

 * igniteui-angular version: 8.1
 * browser: ALL

## Steps to reproduce  

1. Start samples
2. Load "Tabs Routing" sample
(sample loads with relative address /samples/layouts/tabs-sample-6)
3. Click a tab item - for example the second tab
(the address now should be /samples/layouts/tabs-sample-6/departures)
The second tab has a blue icon and text foreground and the blue selection dash below the button is visible.
4. Navigate to the root sample URL by selecting it from the sample's igx-select component (the first item).

## Result  
The content of the router-outlet is empty which corresponds to the URL.
There is no selected tab item (in terms of icon and text highlight).
However, the blue selection dash is left as it was in the previous selection.

## Expected result  
The blue selection dash should not be visible in accordance with the current selection which is "no selected tab item"

## Attachments  
image

### PR DESCRIPTION
**Pull request by rmkrmk**
Closes #5471

### Additional information (check all that apply):
 - [x] Bug fix
 - [ ] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [x] This PR includes unit tests covering all the new code
 - [ ] This PR includes API docs for newly added methods/properties
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/src/lib/tabs/tabs.common.ts [hunk 1]
```diff
@@ -15,15 +15,22 @@ export abstract class IgxTabsBase {
...
+    transformContentAnimation(tab: IgxTabItemBase, duration: number) {}
+    transformIndicatorAnimation(element: HTMLElement) {}
 }
 
 /** @hidden */
-export interface IgxTabItemBase {
+export abstract class IgxTabItemBase {
     nativeTabItem: ElementRef;
-    select(focusDelay?: number);
+    _isSelected = false;
```
Comment (DiyanDimitrov) [inline projects/igniteui-angular/src/lib/tabs/tabs.common.ts]: Replace `_isSelected` field with something like `setSelectedInternal` method for both tab and group base classes.

File: projects/igniteui-angular/src/lib/tabs/tabs.common.ts [hunk 2]
```diff
@@ -15,15 +15,22 @@ export abstract class IgxTabsBase {
...
+    tabindex;
+    select(focusDelay?: number): void {}
```
Comment (DiyanDimitrov) [inline projects/igniteui-angular/src/lib/tabs/tabs.common.ts]: Remove `focusDelay` parameter of the `select` method for both tab and group base classes.

File: projects/igniteui-angular/src/lib/tabs/tabs.component.ts [hunk 3]
```diff
@@ -422,6 +358,105 @@ export class IgxTabsComponent implements IgxTabsBase, AfterViewInit, OnDestroy {
...
+        if (oldTabRelatedGroup) {
+            oldTabRelatedGroup._isSelected = false;
+        }
+        this._selectedIndex = -1;
+        this.onTabItemDeselected.emit({ tab: oldTab, group: oldTabRelatedGroup });
+    }
+
+    private performSelection(newTab: IgxTabItemBase): void {
+        newTab._isSelected = true;
+        newTab.tabindex = 0;
```
Comment (DiyanDimitrov) [inline projects/igniteui-angular/src/lib/tabs/tabs.component.ts]: `tabindex` should be set to -1 on deselection.

Comment (rmkrmk) [reply projects/igniteui-angular/src/lib/tabs/tabs.component.ts]: Addressed.

File: projects/igniteui-angular/src/lib/tabs/tabs.common.ts (hunk 2 above)
Comment (rmkrmk) [reply projects/igniteui-angular/src/lib/tabs/tabs.common.ts]: Addressed.

File: projects/igniteui-angular/src/lib/tabs/tabs.common.ts (hunk 1 above)
Comment (rmkrmk) [reply projects/igniteui-angular/src/lib/tabs/tabs.common.ts]: Addressed.

**Pull request merged**